src/executor/graph_executor.cc

```diff
@@ -643,18 +640,11 @@ void GraphExecutor::InitCachedOps() {
       all_vars.push_back(nd.var());
       mutate_vars.push_back(nd.var());
     }
-    auto dedup = [] (std::vector<Engine::VarHandle>& vars) {  // NOLINT(*)
-      std::sort(vars.begin(), vars.end());
-      vars.resize(std::unique(vars.begin(), vars.end()) - vars.begin());
-    };
-    dedup(use_vars);
-    for (auto v : use_vars) {
-      if (std::binary_search(mutate_vars.begin(), mutate_vars.end(), v)) {
-        LOG(FATAL) << "var duplication happens for op " << inode.source->attrs.name;
-      }
-    }
-    dedup(mutate_vars);
-    dedup(all_vars);
+    // dedup vars
+    Engine::Get()->DeduplicateVarHandle(&use_vars, &mutate_vars);
+    Engine::Get()->DeduplicateVarHandle(&use_vars, &inplace_vars);
+    Engine::Get()->DeduplicateVarHandle(&all_vars, &inplace_vars);
```

Comment:
I think the purpose of inplace_inputs was to avoid pushing vars twice by ignoring outputs that are already in inputs. Since we are doing deduplication here there is not need for keeping inplace_vars any more